src/osx/webview_webkit.mm

```diff
@@ -101,75 +98,64 @@ @interface WebViewCustomProtocol : NSURLProtocol
                                  const wxSize& size, long style,
                                  const wxString& name)
 {
-    m_busy = false;
-    m_nextNavigationIsNewWindow = false;
-
     DontCreatePeer();
     wxControl::Create(parent, winID, pos, size, style, wxDefaultValidator, name);
 
-#if wxOSX_USE_IPHONE
-    CGRect r = wxOSXGetFrameForControl( this, pos , size ) ;
-    m_webView = [[UIWebView alloc] initWithFrame:r];
-    
-    SetPeer( new wxWidgetIPhoneImpl( this, m_webView ) );
-
-#else
     NSRect r = wxOSXGetFrameForControl( this, pos , size ) ;
-    m_webView = [[WebView alloc] initWithFrame:r
-                                     frameName:@"webkitFrame"
-                                     groupName:@"webkitGroup"];
+    WKWebViewConfiguration* webViewConfig = [[WKWebViewConfiguration alloc] init];
+    if (!m_handlers.empty())
+    {
+#if __MAC_OS_X_VERSION_MAX_ALLOWED >= MAC_OS_X_VERSION_10_13
+        if ( WX_IS_MACOS_AVAILABLE(10, 13) )
+        {
+            for (wxStringToWebHandlerMap::iterator it = m_handlers.begin(); it != m_handlers.end(); it++)
+            {
+                [webViewConfig setURLSchemeHandler:[[WebViewCustomProtocol alloc] initWithHandler:it->second.get()]
+                                            forURLScheme:wxCFStringRef(it->first).AsNSString()];
+            }
+        }
+        else
+#endif // macOS 10.13+
+            wxLogDebug("Registering handlers is not supported");
```

Comment:
Sorry for nitpicking, but I think this error may be surprising if you just see it on the console without any context, so I'd like to make the message more precise, e.g.:
```suggestion
            wxLogDebug("Registering custom wxWebView handlers is not supported under macOS < 10.13");
```